### PR TITLE
EPMLABSBRN-823 added fields into Excercise entiy: active , changedBy, changedWhen

### PR DESCRIPTION
**Pull request by andrsam**
[EPMLABSBRN-823](https://jira.epam.com/jira/browse/EPMLABSBRN-823)

**Description**:

### CODE REVIEW

File: src/main/kotlin/com/epam/brn/upload/csv/series4/SeriesFourRecordProcessor.kt [hunk 1]
```diff
@@ -103,7 +102,10 @@ class SeriesFourRecordProcessor(
             name = record.exerciseName,
             level = record.level,
             noiseLevel = record.noiseLevel,
-            noiseUrl = if (!record.noiseUrl.isNullOrEmpty()) String.format(fonAudioPath, record.noiseUrl) else ""
+            noiseUrl = if (!record.noiseUrl.isNullOrEmpty()) String.format(fonAudioPath, record.noiseUrl) else "",
+            active = true,
+            changedBy = userAccountService.getUserFromTheCurrentSession().name,
+            changedWhen = LocalDateTime.now()
```
Comment (ElenaSpb) [inline src/main/kotlin/com/epam/brn/upload/csv/series4/SeriesFourRecordProcessor.kt:108]: please use TimeService

File: src/main/resources/db/migration/V220210519_823.sql [hunk 2]
```diff
@@ -0,0 +1,6 @@
+alter table exercise
+    add column active boolean not null default true;
+alter table exercise
+    add column changed_when timestamp not null default current_timestamp;
+alter table exercise
+    add column changed_by varchar(255) not null default '';
```
Comment (ElenaSpb) [inline src/main/resources/db/migration/V220210519_823.sql:6]: 1. let default be "InitialDataLoader"
2. lets add script to update all current exercises with values
active=true
changed_when=today
changed_by=InitialDataLoader

File: src/main/kotlin/com/epam/brn/config/JpaAuditConfig.kt [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        return DateTimeProvider {
+            Optional.of(dateTimeService.now())
+        }
+    }
+
+    @Bean
+    fun userContextProvider(userAccountService: UserAccountService): AuditorAware<String> {
+        return AuditorAware<String> {
+            val authentication = SecurityContextHolder.getContext().authentication
+            Optional.of(if (authentication != null) authentication.name else "Unauthorized")
```
Comment (ElenaSpb) [inline src/main/kotlin/com/epam/brn/config/JpaAuditConfig.kt:31]: please change Unauthorized - > InitialDataLoader

File: src/main/kotlin/com/epam/brn/dto/ExerciseDto.kt [hunk 4]
```diff
@@ -9,5 +11,8 @@ data class ExerciseDto(
     var template: String? = "",
     var available: Boolean = true,
     var tasks: MutableList<ShortTaskDto> = mutableListOf(),
-    var signals: MutableList<SignalTaskDto> = mutableListOf()
+    var signals: MutableList<SignalTaskDto> = mutableListOf(),
+    var active: Boolean = true,
+    var changedBy: String? = "",
```
Comment (ElenaSpb) [inline src/main/kotlin/com/epam/brn/dto/ExerciseDto.kt:16]: var changedBy: String? = "InitialDataLoader",

File: src/main/resources/db/migration/V220210519_823.sql (hunk 2 above)
Comment (andrsam) [reply src/main/resources/db/migration/V220210519_823.sql]: default vaue was added

File: src/main/kotlin/com/epam/brn/dto/ExerciseDto.kt (hunk 4 above)
Comment (andrsam) [reply src/main/kotlin/com/epam/brn/dto/ExerciseDto.kt]: default vaue was added

File: src/main/resources/db/migration/V220210519_823.sql [hunk 5]
```diff
@@ -0,0 +1,6 @@
+alter table exercise
+    add column active boolean not null default true;
+alter table exercise
+    add column changed_when timestamp not null default current_timestamp;
+alter table exercise
+    add column changed_by varchar(255) not null default 'InitialDataLoader';
```
Comment (ElenaSpb) [inline src/main/resources/db/migration/V220210519_823.sql:6]: please add script to init this new fields in exist entities

File: src/main/kotlin/com/epam/brn/dto/ExerciseDto.kt [hunk 6]
```diff
@@ -9,5 +11,8 @@ data class ExerciseDto(
     var template: String? = "",
     var available: Boolean = true,
     var tasks: MutableList<ShortTaskDto> = mutableListOf(),
-    var signals: MutableList<SignalTaskDto> = mutableListOf()
+    var signals: MutableList<SignalTaskDto> = mutableListOf(),
+    var active: Boolean = true,
+    var changedBy: String? = "InitialDataLoader",
+    var changedWhen: LocalDateTime = LocalDateTime.now()
```
Comment (ElenaSpb) [inline src/main/kotlin/com/epam/brn/dto/ExerciseDto.kt:17]: when we change dto , please do not forget to change api in raml.api

Comment (ElenaSpb) [reply src/main/kotlin/com/epam/brn/dto/ExerciseDto.kt:17]: in https://github.com/Brain-up/brn/tree/master/api-contract/samples

File: src/main/kotlin/com/epam/brn/config/JpaAuditConfig.kt [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
```
Comment (ElenaSpb) [inline src/main/kotlin/com/epam/brn/config/JpaAuditConfig.kt:31]: let it be Unauthorized -> InitialDataLoader

File: src/main/resources/db/migration/V220210519_823.sql (hunk 2 above)
Comment (andrsam) [reply src/main/resources/db/migration/V220210519_823.sql]: update script added

File: src/main/kotlin/com/epam/brn/dto/ExerciseDto.kt (hunk 4 above)
Comment (andrsam) [reply src/main/kotlin/com/epam/brn/dto/ExerciseDto.kt]: update script added

File: src/main/resources/db/migration/V220210519_823.sql (hunk 5 above)
Comment (andrsam) [reply src/main/resources/db/migration/V220210519_823.sql]: script added

**Review by ElenaSpb — APPROVED** [review]

**Pull request merged**